File: Maths_And_Stats/Number_Theory/Armstrong_Number/armstrongnumber.js
// program to check an Armstrong number of n digits

// take an input
let number = prompt("Enter a positive integer");
let numberOfDigits = number.length;
let sum = 0;

// create a temporary variable
let temp = number;

while (temp > 0) {

    let remainder = temp % 10;

    sum += remainder ** numberOfDigits;

    // removing last digit from the number
    temp = parseInt(temp / 10); // convert float into integer
}

if (sum == number) {
    console.log(`${number} is an Armstrong number`);
} else {
    console.log(`${number} is not an Armstrong number.`);
}

File: Maths_And_Stats/Number_Theory/Armstrong_Number/Armstrong_Number.c
#include <iostream>

using namespace std;

int main()
{
    //The number of test cases
    cout << "Enter the number of testcases" << endl;
    int t;
    scanf("%d", &t);
    while(t--){
        int n,rem,ans=0;
        printf("Enter a Number: ");
        scanf("%d", &c);

        int n1=n;
        while(n1>0){
            rem = n1%10;
            ans = ans + (rem*rem*rem);
            n1 = n1/10;
        }

        if(ans == n)
             printf("The number %d is an armstrong number", n);
        else
            printf("The number %d is not an armstrong number", n);
    }
